WorldSegment.h

```diff
@@ -46,7 +46,11 @@ class WorldSegment
 
         uint32_t newNumTiles = inState.Size.x * inState.Size.y * inState.Size.z;
         uint32_t memoryNeeded = newNumTiles * sizeof(Tile);
-        tiles = new Tile[newNumTiles];
+        tiles = (Tile *)new char[memoryNeeded];
+        //memset(tiles, 0, memoryNeeded);
+        for(uint32_t i = 0; i < newNumTiles; i++) {
+            tiles[i].valid = false;
```

Comment:
Another reason I'd like to switch towards allocating/deallocating `Tile`s instead of managing memory manually: the default `Tile` constructor already sets `valid` to `false`, so this logic (also present below) would be unnecessary.

Reply:
I was just changing this to (mostly) match its original functionality before the commit that broke it - changing it to allocate/deallocate individual `Tile`s might result in severe performance penalties (which is probably why it was done this way in the first place).